### ISSUE TITLE
Project Feedback

### ISSUE THREAD

**Issue by woodsbr6** [woodsbr6/MI-449-SS17-740-js-localstorage-wMUDce#1]
## Build a website that allows the user to switch between day and night color themes and remembers their choice

@KatieMFritz Can you take a look at this? It's [hosted here](https://woodsbr6-kapstpq2uw6ovwmudce.surge.sh/) and meets the following criteria:

- [x] Includes a button that when clicked, will toggle the website between day and night themes
- [x] When a user changes the theme, their choice is remembered in `localStorage`
- [x] The class on the `body` element is always up-to-date with the current theme (either `day-theme` or `night-theme`) -- you can use `document.body.setAttribute('class', 'some-class-name')` to set the class on the `body`
- [x] When the body has the class `day-theme`, the background of the website is light-colored, with a darker color for the text
- [x] When the body has the class `night-theme`, the background of the website is a very dark color, with a lighter color for the text
- [x] Somewhere on the page, the user is told how many times they've visited the website (starting with `1` the first time)

**Comment by stuartpearman** [comment]
Hey Brielle, I see you have the button working on github pages, make sure you push to surge using the command in the project section of the lesson.

Also, see if you can get this working with localStorage. Follow along with the lesson and let me know if you get stumped or have questions.

**Comment by woodsbr6** [comment]
@stuartpearman hey yeah im having some issues i cant seem to figure out what to assign, i put some form of code in and then it says that its assigned but never used and idk where im going wrong

**Comment by stuartpearman** [comment]
Can you push what you have up to GitHub?

**Comment by woodsbr6** [comment]
ok its there i didnt see u responded

**Comment by woodsbr6** [comment]
@egillespie

**Comment by stuartpearman** [comment]
So you're getting the property `'nightlight'` from localStorage but you're never setting it with the `window.localStorage.setItem()` function.

Once you have that in place, you'll need to actually use the variable you get from localStorage to set the page to day mode or night mode.

Right now you're also using `style.color` and `style.backgroundColor`, but it would be better to set the class and style your day and night themes with CSS. See criteria 3:

>  The class on the body element is always up-to-date with the current theme (either day-theme or night-theme) -- you can use document.body.setAttribute('class', 'some-class-name') to set the class on the body

Then you could save the name of the class to localStorage and simplify things a bit.

**Comment by stuartpearman** [comment]
I would also make sure to re-read the first slide all the way through and see if that clarifies things a bit:

https://msu.lansing.codes/courses/MI-449-SS17-740/lessons/js-localstorage/1

**Comment by woodsbr6** [comment]
ok i made some changes

**Comment by stuartpearman** [comment]
Not really sure what to do with this, I understand you're stumped but I don't want to just give you the answers either.

Feel free to ask questions of course, but you only took 10 minutes on those last changes. Half of the things I asked you to adjust weren't even attempted.

Was my last comment confusing or unhelpful?

**Comment by woodsbr6** [comment]
yes it was kind of confusing so i looked at the link u gave me and applied that code and tried to change the names for my code, nightlight for the local storage and saying if nightlight is null that means that the color background will be night (black) and if that is null then the background color is day white

maybe im not understanding completely but thats what i was taking from it 

var nightLight = window.localStorage.getItem('nightLight')

   // If no firstName was previously saved...
   if (nightLight === null) {
     // Ask the user for their first name
     color = window.prompt('Night!')
     if (color === null) {
       backgroundColor = 'Day'
     } else {
       window.localStorage.setItem('Night', 'Day')
     }
   }

**Comment by stuartpearman** [comment]
So `nightLight` can be the variable that we use to save the body element's classname. When you check to see if nightLight is null, you're checking to see if there is a class that was previously saved. If it's null, rather than setting the value of a `color` variable, set the value of the `nightlight` variable.

Once that's done, you can set the body class to whatever class is saved to the `nightlight` variable, and use CSS to style the element with that classname.

And since we want to save our class each time it's changed, we should use `window.localStorage.setItem()` at the end of the `toggleNightMode` function.

**Comment by woodsbr6** [comment]
ok so i change the variables to nightlight, and at the end i change localstorage.setitem to togglenightmode?

**Comment by stuartpearman** [comment]
I'm not really sure what you mean. Maybe we should do a video call and try to help you figure it out? Message me on Slack

**Comment by stuartpearman** [comment]
:+1: :shipit:
